core/src/androidTest/java/org/hisp/dhis/android/core/category/CategoryCategoryOptionLinkStoreShould.java

```diff
@@ -34,52 +41,111 @@ public void setUp() throws IOException {
+
+        thenAssertThereAreNotItemsOnCategoryOptionLinkTable();
+    }
+
+    @Test
+    public void update_a_category_option_link() {
+
+        whenInsertNewCategoryOptionLink();
+
+        thenAssertThatNewCategoryOptionLinkWasInserted();
+
+        whenUpdateANewCategoryOptionLink();
+
+        thenAssertThatCategoryOptionWasUpdated();
+
+    }
+
+    @Test
+    public void query_all_category_options_link(){
+
+        whenInsertNewCategoryOptionLink();
+
+        thenAssertThatNewCategoryOptionLinkWasInserted();
+
+        thenAssertQueryAllBringData();
+
+    }
+
+    private void thenAssertQueryAllBringData() {
+        List<CategoryCategoryOptionLinkModel> items = store.queryAll();
+        assertFalse(items.isEmpty());
+    }
+
+    private void thenAssertThatCategoryOptionWasUpdated() {
+
+        List<CategoryCategoryOptionLinkModel> list = store.queryAll();
+        assertTrue(Objects.equals(list.get(0).category(), UPDATED_CATEGORY_UID));
+    }
+
+    private void whenDeleteAllElementsOnCategoryComboLinkTable() {
+        store.delete();
+    }
+
+    private void thenAssertThereAreNotItemsOnCategoryOptionLinkTable() {
+        assertTrue(store.queryAll().isEmpty());
+    }
+
+    private void whenDeleteNewCategoryOptionLink() {
+        store.delete(newCategoryCategoryOptionLinkModel);
     }
 
     private void givenACategory() {
-        Date today = new Date();
-
-        newCategory = Category.builder()
-                .uid("KfdsGBcoiCa")
-                .code("BIRTHS_ATTENDED")
-                .created(today)
-                .name("Births attended by")
-                .shortName("Births attended by")
-                .displayName("Births attended by")
-                .categoryOptions(new ArrayList<CategoryOption>())
-                .dataDimensionType("DISAGGREGATION").build();
-    }
-
-    private void givenACategoryOption() {
-        Date today = new Date();
-
-        newCategoryOption = CategoryOption.builder()
-                .uid("TNYQzTHdoxL")
-                .code("MCH_AIDES")
-                .created(today)
-                .name("MCH Aides")
-                .shortName("MCH Aides")
-                .displayName("MCH Aides")
-                .build();
+        newCategory = generateCategory(DEFAULT_CATEGORY_UID);
+    }
+
+    private void givenACategoryOption(String uid) {
+        newCategoryOption = generateANewCategoryOption(uid);
+    }
+
+    private CategoryOption generateANewCategoryOption(String uid) {
+        return generatedCategoryOption(uid);
     }
 
     private void givenACategoryOptionLinkModel() {
         newCategoryCategoryOptionLinkModel = CategoryCategoryOptionLinkModel.builder()
-                .option("TNYQzTHdoxL")
-                .category("KfdsGBcoiCa")
+                .categoryOption(DEFAULT_CATEGORY_OPTION_UID)
+                .category(DEFAULT_CATEGORY_UID)
                 .build();
     }
 
     private void whenInsertNewCategoryOptionLink() {
+        givenACategory();
```

Comment:
this method has two responsibilities, create a fixture (givens...) and realize actions (whens..), I think that the code would be more readable in tests where is used if split this method in two, one responsibility per method, can you refactor it?

Reply:
Sure!